app/components/banner/hooks/useBannerAnimation.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import type {BannerPosition} from '../Banner';
+
+interface UseBannerAnimationProps {
+    visible: boolean;
+    position: BannerPosition;
+    animationDuration: number;
+}
+
+export const useBannerAnimation = ({
+    visible,
+    position,
+    animationDuration,
+}: UseBannerAnimationProps) => {
+    const opacity = useSharedValue(visible ? 1 : 0);
+    const slideOffset = position === 'top' ? -50 : 50;
+    const translateY = useSharedValue(visible ? 0 : slideOffset);
+    const translateX = useSharedValue(0);
+    const isDismissed = useSharedValue(false);
+
+    const animatedStyle = useAnimatedStyle(() => ({
+        opacity: withTiming(opacity.value, {duration: animationDuration}),
+        transform: [
+            {
+                translateY: withTiming(translateY.value, {duration: animationDuration}),
+            },
+            {
+                translateX: translateX.value,
+            },
+        ],
+    }));
+
+    React.useEffect(() => {
+        if (!isDismissed.value) {
+            opacity.value = visible ? 1 : 0;
+            const offset = position === 'top' ? -50 : 50;
+            translateY.value = visible ? 0 : offset;
+            translateX.value = 0;
+        }
+    }, [visible, position, opacity, translateY, translateX, isDismissed]);
```

Comment:
probably isDismissed's `.value` should be what you're depending on, right? Then you should also depend on the values of the other objects as well.

Reply:
I'm removing everything except for visible and position and adding eslint-disable-line.  The rest are react-native-reanimated shared value, and they will not cause a re-render.

> When you change the sv.value Reanimated will update the styles and keep the shared value in sync between the threads. However, this won't trigger a typical React re-render because a shared value is a plain JavaScript object.